File: Notes/Notes/Validators/Validator.cs
using System;
    /// <summary>
    /// Класс с универсальными методами проверки значений.
    /// </summary>
public static class Validator
{
    /// <summary>
    /// Проверка строки на наличие символов.
    /// </summary>
    /// <param name="value">Проверяемая строка.</param>
    /// <param name="propertyName">Имя объекта, которое подлежит проверке.</param>
    public static void AssertNotEmpty(string value, string propertyName)
    {
        if (string.IsNullOrWhiteSpace(value))
            throw new ArgumentException($"Свойство {propertyName} не может быть пустым.");
    }

    /// <summary>
    /// Проверка строки на количество символов.
    /// </summary>
    /// <param name="value">Проверяемая строка.</param>
    /// <param name="maxLength">Максимальное значение.</param>
    /// <param name="propertyName">Имя объекта, которое подлежит проверке.</param>
    public static void AssertMaxLength(string value, int maxLength, string propertyName)
    {
        if (value != null && value.Length > maxLength)
            throw new ArgumentException($"Свойство {propertyName} не должно превышать {maxLength} символов.");
    }
}

File: DSlab3/DSlab3/Validator.cpp
#include <iostream>
#include "Validator.h"

/// <summary>
/// Получает валидный числовой ввод от пользователя.
/// </summary>
/// <param name="prompt">Сообщение для пользователя</param>
/// <returns>Введенное число</returns>
int GetValidatedInput(const std::string& prompt)
{
    int value;
    while (true)
    {
        std::cout << prompt;
        std::cin >> value;

        if (std::cin.fail())
        {
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
            std::cout << "Invalid input. Please enter a number." << std::endl;
        }
        else
        {
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
            return value;
        }
    }
}